source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/events.py

```
@@ -31,6 +33,104 @@
+
+    # acquire stage
+    stage = omni.usd.get_context().get_stage()
+    # resolve prim paths for spawning and cloning
+    prim_paths = sim_utils.find_matching_prim_paths(asset.cfg.prim_path)
+
+    # sample values
+    rand_samples = math_utils.sample_uniform(0.0, 1.0, (len(env_ids), 3), device="cpu").tolist()
+
+    # use sdf changeblock for faster processing of USD properties
+    with Sdf.ChangeBlock():
+        for i, env_id in enumerate(env_ids):
+            # spawn single instance
+            prim_spec = Sdf.CreatePrimInLayer(stage.GetRootLayer(), prim_paths[env_id])
+
+            # get the attribute to randomize
+            color_spec = prim_spec.GetAttributeAtPath(
+                prim_paths[env_id] + "/geometry/material/Shader.inputs:diffuseColor"
+            )
+            color_spec.default = Gf.Vec3f(*rand_samples[i])
+
+
+def randomize_scale(
+    env: ManagerBasedEnv,
+    env_ids: torch.Tensor | None,
+    scale_range: tuple[float, float] | dict[str, tuple[float, float]],
+    asset_cfg: SceneEntityCfg,
+):
+    """Randomize the scale of an asset along all dimensions.
+
+    This function modifies the "xformOp:scale" property of all the prims corresponding to the asset.
+
+    It takes a tuple or dictionary for the scale ranges. If it is a tuple, then the scaling along
+    individual axis is performed equally. If it is a dictionary, the scaling is independent across each dimension.
+
+    The keys of the dictionary are ``x``, ``y``, and ``z``. The values are tuples of the form ``(min, max)``.
+    If the dictionary does not contain a key, the range is set to one for that axis.
+
+    .. attention::
+        Since this function modifies USD properties, it should only be used before the simulation starts
+        playing. This corresponds to the event mode named "spawn". Using it at simulation time, may lead to
+        unpredictable behaviors.
+
+    .. note::
+        When randomizing the scale of individual assets, please make sure to set
```

Should we just throw an exception if `replicate_physics` is True here?